trunk/SUAVE/Methods/Weights/Buildups/Common/fuselage.py

```diff
@@ -0,0 +1,129 @@
+            max_g_load                  Max Accelerative Load During Flight [Unitless]
+            landing_impact_factor       Maximum Load Multiplier on Landing  [Unitless]
+
+        Outputs:
+
+            weight:                 Estimated Fuselage Mass             [kg]
+    """
+
+#-------------------------------------------------------------------------------
+# Unpack Inputs
+#-------------------------------------------------------------------------------
+
+    fLength = config.fuselages.fuselage.lengths.total
+    fWidth  = config.fuselages.fuselage.width
+    fHeight = config.fuselages.fuselage.heights.maximum
+    maxSpan = config.wings["main_wing"].spans.projected
+    MTOW    = config.mass_properties.max_takeoff
+    G_max   = maximum_g_load
+    LIF     = landing_impact_factor
+    SF      = safety_factor
+
+    # Calculate Skin Weight Per Unit Area (arealWeight) based on material
+    # properties. In this instance we assume the use of
+    # Bi-directional Carbon Fiber, a Honeycomb Core, and Paint:
+
+    arealWeight =(
+          bidirectional_carbon_fiber().minimum_gage_thickness      * bidirectional_carbon_fiber().density
+        + honeycomb().minimum_gage_thickness * honeycomb().density
+        + paint().minimum_gage_thickness     * paint().density
+        )
+
+    # Calculate fuselage area (using assumption of ellipsoid), and weight:
+
+    S_wet = 4 * np.pi * (((fLength * fWidth/4)**1.6
+        + (fLength * fHeight/4)**1.6
+        + (fWidth * fHeight/4)**1.6)/3)**(1/1.6)
+    skinMass = S_wet * arealWeight
+
+    # Calculate the mass of a structural bulkhead
+
+    bulkheadMass = 3 * np.pi * fHeight * fWidth/4 * arealWeight
+
+    # Calculate the mass of a canopy, assuming Acrylic:
+
+    canopyMass = S_wet/8 * acrylic().minimum_gage_thickness * acrylic().density
+
+    # Calculate keel mass needed to carry lifting moment, assuming
+    # Uni-directional Carbon Fiber used to carry load
+
+    L_max       = G_max * MTOW * 9.8 * SF        # Max Lifting Load
+    M_lift      = L_max * fLength/2.       # Max Moment Due to Lift
+    beamWidth   = fWidth/3.                # Allowable Keel Width
+    beamHeight  = fHeight/10.              # Allowable Keel Height
+
+    beamArea    = M_lift * beamHeight/(4*unidirectional_carbon_fiber().ultimate_tensile_strength*(beamHeight/2)**2)
+    massKeel    = beamArea * fLength * unidirectional_carbon_fiber().density
+
+    # Calculate keel mass needed to carry wing bending moment, assuming
+    # thin walled Bi-directional Carbon Fiber used to carry load
+
+    M_bend      = L_max/2 * maxSpan/2                          # Max Bending Moment
+    beamArea    = beamHeight * beamWidth                       # Enclosed Beam Area
+    beamThk     = 0.5 * M_bend/(bidirectional_carbon_fiber().ultimate_shear_strength*beamArea)    # Beam Thickness
+    massKeel   += 2*(beamHeight + beamWidth)*beamThk*bidirectional_carbon_fiber().density
```

Comment:
you should unpack these

Reply:
Changed to unpack of material properties.